src/commands/createFunction/promptStepsV2/promptStepFactory.ts

```diff
@@ -0,0 +1,29 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep } from "@microsoft/vscode-azext-utils";
+import { ResourceType } from "../../../templates/IBindingTemplate";
+import { ParsedInput } from "../../../templates/script/parseScriptTemplatesV2";
+import { FunctionWizardV2Context } from "../FunctionV2WizardContext";
+import { EnumInputStep } from "./EnumInputStep";
+import { ExistingFileStep } from "./ExistingFileStep";
+import { NewFileStep } from "./NewFileStep";
+import { StringInputStep } from "./StringInputStep";
+
+export function promptStepFactory<T extends FunctionWizardV2Context>(input: ParsedInput): AzureWizardPromptStep<T> {
+    switch (input.value) {
+        case 'enum':
+            return new EnumInputStep(input);
+        default:
+            switch (input.resource) {
+                case ResourceType.ExistingFile:
+                    return new ExistingFileStep(input);
+                case ResourceType.NewFile:
+                    return new NewFileStep(input);
+                default:
+                    return new StringInputStep(input);
+            }
+    }
+}
```

Comment:
Nested switch statements make me sad but this one seems pretty much unavoidable

Reply:
It's either that or a big if-else block

Reply:
Yeah, and I don't think I like that more

Reply:
![image](https://github.com/microsoft/vscode-azurefunctions/assets/5290572/cae5a724-96e8-4db9-99ad-43e400fa7230)